### DISCUSSION TITLE
Adding broadcasting algorithms and related graph generators

### DISCUSSION THREAD

**Discussion by Transurgeon** [networkx/networkx#6824] [Algorithms]
Hello NetworkX developers,

I had the pleasure of meeting some of you at Scipy 2023 in Austin, Texas, where I first brought this idea up. 

I wanted to propose a project for two undergraduate students (myself included). 
The idea is to add new algorithms in the field of broadcasting to your already existing suite of algorithms. 
Please find a copy of my proposal below.
https://docs.google.com/document/d/15vYHZeLh7x6BWRDYSXd05aOR1Nx0tIzyKvHyhDbpKg8/edit 

I wanted to start a discussion about the steps towards adding a new algorithm to networkx, we can discuss about how to name APIs and other related topics/concerns on your end.

Could you point us to some relatively new algorithms that have been added? it would be nice to have as a reference for good practices.

**Comment by dschult** [comment]
Can you briefly explain how the broadcast time differs from the [eccentricity](https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.distance_measures.eccentricity.html) of a network?  And it's fine if it is the same -- but I want to understand the question that "broadcasting" is addressing in relation to other known problems.

For a recent addition, look at #6633 and the `girth` function.  That added a new function to an existing module. Adding a new module with multiple functions is another approach and connecting the docs to the doc_string would be slightly different as would updating the `__init__.py` file. but I think it gives a good example of what is needed. :)

**Comment by Transurgeon** [comment]
Broadcasting is the time it takes to reach all the nodes in the graph, you are right that there are many similarities with the eccentricity of a network, especially in less connected graphs. 

However, they are still quite different. Just as a simple example, in a fully connected graph with n vertices, the eccentricity is always 1. However, the broadcast of the same graph is: $b(K_n) = \lceil \log_2{n} \rceil$ since every discrete unit of time, the number of vertices that have received the message can at most double. 

I feel like it's possible we could have enough material to create a new module, we will get back to you after discussing about design ideas in more detail.

**Comment by rossbar** [comment]
This was added in #6928 so closing the discussion, thanks all!

**Comment by hvanhemmen** [comment]
Hey @rossbar  and @Transurgeon , are these functions still WIP? I'm trying to use them on a weighted network (non-directional, and I believe it qualifies as a tree). broadcasting.py is erroring out pretty much every time on the min function. Wondering if you're still intending to support this work, and if there are plans for these types of functions to be released officially at any point.

**Reply by rossbar** [reply]
The broadcasting functions are included in NX 3.3 so they are "supported". If you're having issues, please open an issue with a minimal reproducer so we can chase down the problem!
